src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -476,13 +478,13 @@ export class ItemListLayout<I extends ItemObject> extends React.Component<ItemLi
   }
 
   showColumn({ id: columnId, showWithColumn }: TableCellProps): boolean {
-    const { tableId, isConfigurable } = this.props;
+    const { tableId, hiddenColumns, isConfigurable } = this.props;
 
-    return !isConfigurable || !UserStore.getInstance().isTableColumnHidden(tableId, columnId, showWithColumn);
+    return !isConfigurable || !(UserStore.getInstance().isTableColumnHidden(tableId, columnId, showWithColumn) !== hiddenColumns.includes(columnId));
```

Comment:
This seems to imply that a stored value of `true` means something different for a column that has been marked as `hidden`.

That will certainly trip someone up. I think I better check would be to pass `hiddenColumns` to `isTableColumnHidden`. And if it isn't set then return the computed value from the hidden columns list.

That way, it truely acts as a default. The other benefit of this is that we don't have to add a migration or change the user-store.

Reply:
I agree that having to add a migration is not the ideal solution, but then the name `hiddenTableColumns` is also confusing if there is an entry for `qos` lets say, you would think its hidden just by looking at the saved state, when in fact, its now shown because its default is hidden.

I also agree my implementation with the booleans is a bit confusing and requires a migration, so I'll try what you proposed.

Reply:
Good point... maybe a combination? Having a migration is not the end of the world. We just need one with this current change set. And I admit that it would be good for a user to be able to override the default.

Reply:
Ok I'll update to pass hiddenColumns to `isTableColumnHidden`. How would you update the func signature? It currently is `isTableColumnHidden(tableId: string, ...columnIds: string[])` and it feels weird to pass hiddenColumns on either side of tableId.

What about a method in item-list-layout that does the conditional instead?

Reply:
A method in `ItemListLayout` probably would be better. But I would change the return type of `isTableColumnHidden` to `boolean | undefined` and return `undefined` in the cases where the fields don't exist.

Reply:
gotcha, will update :)

Reply:
Also Im not sure how I can create a migration? Do you have docs or maybe just an example I can get inspired from? Or you'd prefer to deal with this yourself?

Thanks

Reply:
There already exist migrations in the `src/migrations/user-store` folder. I would follow their lead. You need to specify a version (I would pick `5.1.4-beta.1` even though we don't plan on releasing a version with that number).